src/test/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusinessIntegrationSpec.scala

```diff
@@ -9,50 +9,56 @@ import org.scalatest.{BeforeAndAfterAll, DoNotDiscover, FlatSpec}
 class AgoraBusinessIntegrationSpec extends FlatSpec with BeforeAndAfterAll with AgoraTestFixture  {
   override protected def beforeAll() = {
     ensureDatabasesAreRunning()
+    startMockWaas()
   }
 
   override protected def afterAll() = {
     clearDatabases()
+    stopMockWaas()
   }
 
   "Agora" should "be able to store a task configuration with a valid (extant) official docker image" in {
-    val entity = patiently(agoraBusiness.insert(testAgoraEntityWithValidOfficialDockerImageInWdl, mockAuthenticatedOwner.get)).addManagers(Seq(mockAuthenticatedOwner.get)).addIsPublic(false)
+    setSingleMockWaasDescribeOkResponse(genericDockerPayloadDescribeResponse)
+
+    val entity = patiently(agoraBusiness.insert(testAgoraEntityWithValidOfficialDockerImageInWdl, mockAuthenticatedOwner.get, mockAccessToken)).addManagers(Seq(mockAuthenticatedOwner.get)).addIsPublic(false)
```

Comment:
here - and in other tests - I'm worried about the usage of `setSingleMockWaasDescribeOkResponse`, which relies on mockserver's `Times.exactly(1)`. Since the mockserver instance is shared across all unit tests in a spec, and the  `Times.exactly(1)` is stateful, this means that:
* if we move to running the unit tests inside a spec in parallel, the mockserver responses are not threadsafe
* if an individual test fails/aborts before making its call to mockserver, the test will not consume its one-time call, and mockserver will return something unexpected to the next unit test

as a quick fix, you could use `beforeEach`/`afterEach` to set mockserver to a pristine state for this api, and add a comment that the spec should never be parallelized.

More ideal would be to set up the mockserver statelessly - have it inspect the payload or some other input and deterministically return success/failure based on what the caller sent (it doesn't have to actually validate; it can just do e.g. a substring lookup)

Reply:
still looking at the other feedback (thanks!) but I wanted to comment on this one since I actually was doing something along these lines tonight before I saw your feedback!  I was annoyed by this as well, so managed to upgrade mockserver (was a pain) so I could use a substring expectation and set up nearly all the requests/responses globally.  A real WaaS would be deterministic (ie the same payload should return the same response) and this just parallels that.

Reply:
I just pushed that code BTW if you want to take a peek